Events/PinwheelEventType.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  PinwheelEvent.swift
+//  PinwheelSDK
+//
+//  Created by Robby Abaya on 2/11/21.
+//
+
+import Foundation
+
+public enum PinwheelEventType: String {
+    case open
```

Comment:
Could we restrict PinwheelEventType to `login`, `exit`, `success`, `error`, and finally `generic` as catch all for every other event? This would allow us to add events on the JavaScript side without requiring developers to update their SDKS. In this case, we would have something like:

```swift
public struct PinwheelGenericEvent: Codable {
  public let type: String
  public let type: Dict<String, Any>
}
```

Reply:
We could do this, but then dealing with the event payloads for the developer would be a lot more cumbersome—casting, validating, etc. And if they wanted to consume the new event, they’d have to update their app anyway. It’s trivial for us to add new events, and deal with translating the payloads once, making it easier and less error prone for everyone—that code completion goes a long way! A middle ground could be to keep all the ones we have, keep the SDKs as updated as possible, but also have a generic handler as a catch-all. Either way, I can make the change, let me know what you think.

Also, the android sdk is implemented in a similar way, so we’d probably want to consider changing it there. cc @TrekSoft

Reply:
I like this middle ground idea! Let's just add the generic handler as a catch all.

Reply:
Ok, resolved this over chat. The decision is to stick with hard types.